fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsManager.java

```diff
@@ -133,7 +139,25 @@ private Table validateTableName(TableName dbTableName) throws AnalysisException
         String tableName = dbTableName.getTbl();
 
         Database db = Catalog.getCurrentCatalog().getDbOrAnalysisException(dbName);
-        Table table = db.getTableOrAnalysisException(tableName);
-        return table;
+        return db.getTableOrAnalysisException(tableName);
+    }
+
+    public void alterTableStatistics(StatisticsTaskResult taskResult) throws AnalysisException {
+        StatsGranularityDesc desc = taskResult.getGranularityDesc();
+        long tableId = desc.getTableId();
+        Map<String, String> statsNameToValue = taskResult.getStatsNameToValue();
+        this.statistics.updateTableStats(tableId, statsNameToValue);
+    }
+
+    public void alterColumnStatistics(StatisticsTaskResult taskResult) throws AnalysisException {
+        StatsCategoryDesc categoryDesc = taskResult.getCategoryDesc();
+        long dbId = categoryDesc.getDbId();
+        long tableId = categoryDesc.getTableId();
+        Database db = Catalog.getCurrentCatalog().getDbOrAnalysisException(dbId);
+        Table table = db.getTableOrAnalysisException(tableId);
+        String columnName = categoryDesc.getColumnName();
+        Type columnType = table.getColumn(columnName).getType();
+        Map<String, String> statsNameToValue = taskResult.getStatsNameToValue();
+        this.statistics.updateColumnStats(tableId, columnName, columnType, statsNameToValue);
```

Comment:
what will happen, if schema changed when we before this function and after statistics job started?

Reply:
same as above~